### PR TITLE
[Merged by Bors] - chore: simplify proof of IsPiSystem.comap

### PR DESCRIPTION
**Pull request by dwrensha**
Uses `Set.not_nonempty_iff_eq_empty` to shorten the proof of `IsPiSystem.comap`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dwrensha** [comment]
I found this with my [tryTacticAtEachStep](https://github.com/dwrensha/compfiles/blob/28828353b229aa1ef1fdcaa3be2a656f02869ac8/scripts/tryTacticAtEachStep.lean) tool.

File: Mathlib/MeasureTheory/PiSystem.lean [hunk 1]
```diff
@@ -106,10 +106,7 @@ theorem IsPiSystem.comap {α β} {S : Set (Set β)} (h_pi : IsPiSystem S) (f : 
   rintro _ ⟨s, hs_mem, rfl⟩ _ ⟨t, ht_mem, rfl⟩ hst
   rw [← Set.preimage_inter] at hst ⊢
   refine' ⟨s ∩ t, h_pi s hs_mem t ht_mem _, rfl⟩
-  by_contra h
-  rw [Set.not_nonempty_iff_eq_empty] at h
-  rw [h] at hst
-  simp at hst
+  exact nonempty_of_nonempty_preimage hst
```
Comment (Ruben-VandeVelde) [inline Mathlib/MeasureTheory/PiSystem.lean:109]: Looks like you can even collapse this into the previous line

Comment (dwrensha) [reply Mathlib/MeasureTheory/PiSystem.lean]: Done in https://github.com/leanprover-community/mathlib4/pull/11052/commits/9b8d5adfce2abede54d70e1444a329c0a337fac3.

**Comment by RemyDegenne** [comment]
Looks good, thanks!
And sorry that it took several weeks since the last review for someone to notice and merge this.
bors r+

**Pull request closed without merging**